### PR TITLE
feat: supports passing user params into coprocessor

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Main changes:

1. Supports passing user params into scripts, for example:

```python
@copr(returns = ["number"])
def test(**params)->vector[i64]:
    return int(params['a']) + int(params['b'])
```

Or coprocessing with a query:

```python
@copr(sql='select uint32s as number from numbers limit 5', args=['number'], returns=['n'])
def test(n, **params)  -> vector[i64]:
    return n + int(params['a'])
```

As above, the `args` and `sql` in the `@coprocessor` annotation are both optional now. 

2. Let `/run-script` API accepts these params by URL params such as `run-script?name=test&a=1&b=1`. The params is in the form of `HashMap<String, String>`, and passed into python script as `KwArgs` such as `**params`.
3. Adds `concat` function to `PyVector` to merge the vectors.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/script/src/python/coprocessor/compile.rs [hunk 1]
```diff
@@ -31,23 +31,40 @@ fn create_located<T>(node: T, loc: Location) -> Located<T> {
 /// generate a call to the coprocessor function
 /// with arguments given in decorator's `args` list
 /// also set in location in source code to `loc`
-fn gen_call(name: &str, deco_args: &DecoratorArgs, loc: &Location) -> ast::Stmt<()> {
```
Comment (discord9) [inline src/script/src/python/coprocessor/compile.rs:34]: Just realize we can just directly use `vm.call_method(obj, method_name, args)` to call methods with `FuncArgs`(Support PosArg&KwArg) manually constructed. Should we change it here or create a new PR?

Comment (killme2008) [reply src/script/src/python/coprocessor/compile.rs:34]: I can try it in this PR.

**Review by discord9 — APPROVED** [review]

File: src/script/src/python/coprocessor/compile.rs [hunk 2]
```diff
@@ -31,23 +31,40 @@ fn create_located<T>(node: T, loc: Location) -> Located<T> {
...
+                create_located(node, loc)
+            })
+            .collect()
+    } else {
+        vec![]
+    };
+
+    if let Some(kwarg) = kwarg {
+        let node = ast::ExprKind::Name {
+            id: kwarg.to_owned(),
```
Comment (evenyag) [inline src/script/src/python/coprocessor/compile.rs:62]: Could we just use clone
```suggestion
                    id: v.clone(),
                    ctx: ast::ExprContext::Load,
                };
                create_located(node, loc)
            })
            .collect()
    } else {
        vec![]
    };

    if let Some(kwarg) = kwarg {
        let node = ast::ExprKind::Name {
            id: kwarg.clone(),
```

File: src/script/src/python/coprocessor/compile.rs [hunk 3]
```diff
@@ -89,6 +111,20 @@ pub fn compile_script(name: &str, deco_args: &DecoratorArgs, script: &str) -> Re
                 type_comment: __main__,
             } = &mut stmt.node
             {
+                // Rewrite kwargs in coprocessor, make it as a positional argument
+                if !decorator_list.is_empty() {
+                    if let Some(kwarg) = kwarg {
+                        args.kwarg = None;
+                        let node = ArgData {
+                            arg: kwarg.to_owned(),
```
Comment (evenyag) [inline src/script/src/python/coprocessor/compile.rs:119]:
```suggestion
                            arg: kwarg.clone(),
```

**Comment by killme2008** [comment]
@evenyag PTAL.  I removed all `to_owned` in `script` crate.

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**